go/libraries/doltcore/sqle/dtables/diff_table.go

```diff
@@ -127,6 +134,7 @@ func (dt *DiffTable) Schema() sql.Schema {
 }
 
 func (dt *DiffTable) Partitions(ctx *sql.Context) (sql.PartitionIter, error) {
+	// TODO: if this were in parallel it can't always start at HEAD
```

Comment:
This call doesn't happen in parallel. We get the partitions, then iterate them in parallel.